returnn/tf/network.py

```diff
@@ -2036,6 +2036,20 @@ def register_search_choices_for_beam(self, beam, search_choices):
       return root_net.register_search_choices_for_beam(beam, search_choices)
     self._map_search_beam_to_search_choices[beam] = search_choices
 
+  def iter_layers_recursively(self):
+    """
+    Yield all sub-layers of this network (including layers inside rec-layers)
+    """
+    from .layers.rec import RecLayer, _SubnetworkRecCell
+    for layer in self.layers.values():
```

Comment:
Do this in a deterministic way (deterministic also in Python <=3.6).
I.e. like:
```suggestion
    for _, layer in sorted(self.layers.items()):
```